### PR TITLE
Configuration

### PR DESCRIPTION
**Pull request by loansindi**
Adds a configuration dialog for the quick buttons.

### CODE REVIEW

**Review by hef — COMMENTED** [review]
git rm --cached *.o
git rm --cached Makefile

File: buttonconfig.cpp [hunk 1]
```diff
@@ -19,6 +21,15 @@ buttonConfig::~buttonConfig()
 
 void buttonConfig::on_buttonBox_accepted()
 {
-    qDebug() << ui->lineEdit_1->text();
     emit(configAccepted(ui->lineEdit_1->text(), ui->lineEdit_2->text(), ui->lineEdit_3->text(), ui->lineEdit_4->text(), ui->lineEdit_5->text(), ui->lineEdit_6->text()));
 }
+
+void buttonConfig::receiveButtons(QString one, QString two, QString three, QString four, QString five, QString six)
```
Comment (hef) [inline buttonconfig.cpp]: This looks like you can only change the buttons at runtime, and they will be reset next time the program is launched.

You may want to checkout the QSettings thing to store settings and load settings at launch: http://doc.qt.io/qt-5/qsettings.html

File: weather-qt.pro.user [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (hef) [inline weather-qt.pro.user]: git rm --cached weather-qt.pro.user

**Review by hef — COMMENTED** [review]
git rm --cached weather-qt
git rm --cached weather-qt.pro.user

File: mainwindow.cpp [hunk 3]
```diff
@@ -174,32 +176,50 @@ void MainWindow::printReport(QString city)
...
 {
-    printReport("KMSP");
+    printReport(ui->pushButton_6->text());
 }
 
-void MainWindow::on_pushButton_KCMX_clicked()
+void MainWindow::on_actionSet_Buttons_triggered()
 {
-    printReport("KCMX");
+    buttonConfig *buttons = new buttonConfig;
```
Comment (hef) [inline mainwindow.cpp]: you should tell buttonConfig to be parented to the main Window:
`buttonConfig *button = new buttonConfig(this);`

Comment (loansindi) [reply mainwindow.cpp]: That explains something I had trouble with.

File: weather-qt.pro.user (hunk 2 above)
Comment (loansindi) [reply weather-qt.pro.user]: I thought i had gotten all these, thanks.

File: buttonconfig.cpp (hunk 1 above)
Comment (loansindi) [reply buttonconfig.cpp]: That's somewhere on my roadmap, yeah.

**Pull request merged**